src/EnergyPlus/SetPointManager.cc

```diff
@@ -6551,6 +6762,200 @@ namespace SetPointManager {
+		
+		// NOTES:
+		// The assumptions related to lagging of setpoint are most suited for smaller timesteps and/or plants that don't vary wildly from one time step to another
+		// The assumptions also become affected by variable flow plants more-so than constant-flow plants
+
+		// Using/Aliasing
+		using namespace DataPlant;
+		using DataLoopNode::Node;
+
+		// we need to know the plant to get the fluid ID in case it is glycol
+		// but we have to wait in case plant isn't initialized yet
+		// if plant isn't initialized, assume index=1 (water)
+		int fluidIndex = 1;
+		if ( this->plantLoopIndex == 0 ) {
+			for ( int plantIndex = 1; plantIndex <= DataPlant::TotNumLoops; plantIndex++ ) {
+				if ( this->supplyNodeIndex == DataPlant::PlantLoop( plantIndex ).LoopSide( 2 ).NodeNumOut ) {
+					this->plantLoopIndex = plantIndex;
+					this->plantSetpointNodeIndex = DataPlant::PlantLoop( plantIndex ).TempSetPointNodeNum;
+					fluidIndex = DataPlant::PlantLoop( plantIndex ).FluidIndex;
+				}
+			}
+		}
+
+		// we don't need fluid names since we have a real index, so just pass in the temperature and get properties
+		Real64 avgTemp = ( returnNode.Temp + supplyNode.Temp ) / 2;
+		Real64 cp = FluidProperties::GetSpecificHeatGlycol( "", avgTemp, fluidIndex, "ReturnWaterChWSetPointManager::calculate" );
+
+		// get the operating flow rate
+		Real64 mdot = supplyNode.MassFlowRate;
+		
+		// calculate the current demand
+		Real64 Qdemand = mdot * cp * ( returnNode.Temp - supplyNode.Temp );
+
+		// check for strange conditions
+		if ( Qdemand < 0 ) {
+			this->currentSupplySetPt = this->minimumChilledWaterSetpoint;
+			return;
+		}
+
+		// Determine a return target, default is to use the constant value, but scheduled or externally 
+		//  set on the return node TempSetPoint will overwrite it.  Note that the schedule index is only
+		//  greater than zero if the input type is scheduled, and the useReturnTempSetpoint flag is only
+		//  true if the input type is specified as such
+		Real64 T_return_target = this->returnTemperatureConstantTarget;
+		if ( this->returnTemperatureScheduleIndex > 0 ) {
+			T_return_target = GetCurrentScheduleValue( this->returnTemperatureScheduleIndex );
+		} else if ( this->useReturnTempSetpoint ) {
+			if ( returnNode.TempSetPoint != -999 ) {
+				T_return_target = returnNode.TempSetPoint;
+			} else {
+				ShowSevereError( "Return temperature reset setpoint manager encountered an error." );
+				ShowContinueError( "The manager is specified to look to the return node setpoint to find a target return temperature, but the node setpoint was invalid" );
+				ShowContinueError( "Verify that a separate sepoint manager is specified to set the setpoint on the return node named \"" + NodeID( this->returnNodeIndex ) + "\"" );
+				ShowContinueError( "Or change the target return temperature input type to constant or scheduled" );
+			}
```

Comment:
This should be fatal.  Other SPMs finish with
  ShowFatalError( "Missing reference setpoint." );